lib/matplotlib/legend.py

```diff
@@ -973,3 +1110,136 @@ def draggable(self, state=None, use_blit=False, update="loc"):
+        # Consider two handles to be the same if they share a label,
+        # color, facecolor, and edgecolor.
+
+        # Loop through each handle and label already collected
+        for f_h, f_l in zip(handles, labels):
+            if f_l != l:
+                continue
+            if type(f_h) != type(h):
+                continue
+            try:
+                if f_h.get_color() != h.get_color():
+                    continue
+            except AttributeError:
+                pass
+            try:
+                if f_h.get_facecolor() != h.get_facecolor():
+                    continue
+            except AttributeError:
+                pass
+            try:
+                if f_h.get_edgecolor() != h.get_edgecolor():
+                    continue
+            except AttributeError:
+                pass
+            return True
+        return False
+
+    for handle in _get_legend_handles(axs, legend_handler_map):
+        label = handle.get_label()
+        if (label
+                and not label.startswith('_')
+                and not _in_handles(handle, label)):
+            handles.append(handle)
+            labels.append(label)
+    return handles, labels
+
+
+def _parse_legend_args(axs, *args, **kwargs):
+    """
+    Get the handles and labels from the calls to either ``figure.legend``
+    or ``axes.legend``.
+
+    ``axs`` is a list of axes (to get legend artists from)
+    """
+
+    handlers = kwargs.get('handler_map', {}) or {}
+
+    # Support handles and labels being passed as keywords.
+    handles = kwargs.pop('handles', None)
+    labels = kwargs.pop('labels', None)
+
+    if (handles is not None or labels is not None) and len(args):
+        warnings.warn("You have mixed positional and keyword "
+                      "arguments, some input may be "
+                      "discarded.")
+
+    # if got both handles and labels as kwargs, make same length
+    if handles and labels:
+        handles, labels = zip(*zip(handles, labels))
+
+    elif handles is not None and labels is None:
+        labels = [handle.get_label() for handle in handles]
+
+    elif labels is not None and handles is None:
+        # Get as many handles as there are labels.
+        handles = [handle for handle, label
+                   in zip(_get_legend_handles(axs, handlers), labels)]
+
+    # No arguments - automatically detect labels and handles.
+    elif len(args) == 0:
+        handles, labels = _get_legend_handles_labels(axs, handlers)
+        if not handles:
+            return None
+
+    # One argument. User defined labels - automatic handle detection.
+    elif len(args) == 1:
+        labels, = args
+        # Get as many handles as there are labels.
+        handles = [handle for handle, label
+                   in zip(_get_legend_handles(axs, handlers), labels)]
+
+    # Two arguments:
+    #   * user defined handles and labels
+    elif len(args) >= 2:
```

Comment:
This should be `==`, passing `loc` as a positional argument in deprecated for the Axes version but apparently we missed doing the same to the Figure version (as it did not explicitly take `loc` as a positional, but passed `*args` through to the `Legend.__init__` so it worked.

This should probably return `handles, labels, extra_args` and in `Axes.legend` raise if `len(extra_args) > 0` and pass them through in `Figure.legend`

Reply:
OK, this is caught in `lib/matplotlib/axes/_axes.py`.    But it still needs to be in here.